src/ovirtapi/types.js

```diff
@@ -47,7 +48,7 @@ export type ApiSnapshotType = {
   snapshot_type?: string,
   date?: number,
   snapshot_status?: string,
-  persist_memorystate?: string
+  persist_memorystate: BoolStringType
```

Comment:
I don't think this is needed. Please check how the NicType.plugged is handled as an example.
BTW, the values for api Boolean type can be either all combinations of upper/lower cases for true/false and also 0/1 as mentioned: http://ovirt.github.io/ovirt-engine-api-model/master/#types/boolean

Reply:
> Represents the false and true concepts used in mathematical logic.
>
>The valid values are the strings false and true.
>
>Case is ignored by the engine, so for example False and FALSE also valid values. However the server will always return lower case values.

The right values for the api are "true" OR "false" strings, any case combination.

> For backwards compatibility with older versions of the engine, the values 0 and 1 are also accepted. The value 0 has the same meaning than false, and 1 has the same meaning than true. Try to avoid using these values, as support for them may be removed in the future.

`NicType.plugged` is an internal types, in web ui we saving boolean as true/false values not strings, but for the API we should convert this value to string value (`"true"` or `"false"` with any combination of lower/upper case).
I don't know why it works now but it probably work because it relies on backward compatibility, it sends true false as values.

The right way to do it is to convert the values of true/false boolean type to strings, and that what I did here, boolean is a primitive type and the server requires string.

Reply:
> `NicType.plugged` is an internal types, in web ui we saving boolean as true/false values not strings, but for the API we should convert this value to string value (`"true"` or `"false"` with any combination of lower/upper case).
> I don't know why it works now but it probably work because it relies on backward compatibility, it sends true false as values.
> 
> The right way to do it is to convert the values of true/false boolean type to strings, and that what I did here, boolean is a primitive type and the server requires string.

It seems to me that we should handle all API Boolean attributes the same way. 
The `Nic.plugged` attribute http://ovirt.github.io/ovirt-engine-api-model/master/#types/nic is just another example for that and it should behave the same as for snapshot.persistMemoryState:
https://github.com/oVirt/ovirt-web-ui/blob/7f65e3ef47106d5028a52e8b2a8c9eeef795aa7a/src/ovirtapi/transform.js#L723
https://github.com/oVirt/ovirt-web-ui/blob/7f65e3ef47106d5028a52e8b2a8c9eeef795aa7a/src/ovirtapi/transform.js#L740

Based on this, the `persist_memorystate` should be declared as Boolean and not String

Reply:
> I don't know why it works now but it probably work because it relies on backward compatibility, it sends true false as values.
> 

Because JSON and the JSON parser should handle it all just fine.  Keep in mind the rest api also supports XML encoding of all of the data, and that handles boolean values differently.

> The right way to do it is to convert the values of true/false boolean type to strings, and that what I did here, boolean is a primitive type and the server requires string.

Based on what evidence?  Does the call fail if the POST body has `persist_memorystate: true` instead of `persist_memorystate: "true"`?